OpenRA.Mods.Common/Traits/RevealsMap.cs

```diff
@@ -51,14 +51,28 @@ protected PPos[] ProjectedCells(Actor self)
 			return self.World.Map.ProjectedCells;
 		}
 
+		void INotifyOwnerChanged.OnOwnerChanged(Actor self, Player oldOwner, Player newOwner)
+		{
+			if (!IsTraitDisabled)
+			{
+				foreach (var player in self.World.Players)
+				{
+					RemoveCellsFromPlayerShroud(self, player);
+					AddCellsToPlayerShroud(self, player, ProjectedCells(self));
```

Comment:
For performance, do `var cells = ProjectedCells(self);` outside the loop, do the same in `TraitEnabled`.

See a similar pattern in `AffectsShroud.UpdateShroudCells` for an example.

Reply:
Done.